sw/device/sca/aes_serial/aes_serial.c

```diff
@@ -0,0 +1,334 @@
+  CHECK(dif_rv_timer_counter_set_enabled(&timer, kHart, kDifRvTimerEnabled) ==
+       kDifRvTimerOk);
+
+  aes_data_put_wait(plain_text);
+  if (dif_gpio_all_write(&gpio, kGpioCaptureTriggerH) != kDifGpioOk) {
+    return kSimpleSerialError;
+  }
+
+  aes_manual_trigger();
+  wait_for_interrupt();
+
+  if (dif_gpio_all_write(&gpio, kGpioCaptureTriggerL) != kDifGpioOk) {
+    return kSimpleSerialError;
+  }
+  aes_data_get_wait(cipher_text);
+
+  print_cmd_response('r', cipher_text, plain_text_len);
+  return kSimpleSerialOk;
+}
+
+/**
+ * Handle simple serial command.
+ *
+ * @param aes_cfg AES configuration object.
+ * @param cmd     input command buffer.
+ * @param cmd_len number of characters set in input buffer.
+ */
+static void simple_serial_handle_command(const aes_cfg_t *aes_cfg,
+                                         const char *cmd, size_t cmd_len) {
+  // Data length is half the size of the hex encoded string.
+  const size_t data_len = cmd_len >> 1;
+  if (data_len >= kMaxInputLengthBin) {
+    return;
+  }
+  char data[64] = {0};
+
+  // The simple serial protocol does not expect return status codes for invalid
+  // data, thus it is ok to return early.
+  if (a2b_hex(&cmd[1], data, data_len) == kSimpleSerialError) {
+    return;
+  }
+
+  if (cmd[0] == 'v') {
+    print_cmd_response('z', (const char *)&kSimpleSerialProtocolVersion,
+                       /*data_len=*/1);
+    return;
+  }
+
+  simple_serial_result_t result = kSimpleSerialOk;
+  switch (cmd[0]) {
+    case 'k':
+      result = simple_serial_set_key(aes_cfg, data, data_len);
+      break;
+    case 'p':
+      result = simple_serial_trigger_encryption(data, data_len);
+      break;
+    case 'v':
+      result = kSimpleSerialProtocolVersion;
+      break;
+  }
+
+  // This protocol version expects a 1 byte return status.
+  print_cmd_response('z', (const char *)&result, /*data_len=*/1);
+}
+
+int main(int argc, char **argv) {
+  uart_init(kUartBaudrate);
+  base_set_stdout(uart_stdout);
+  pinmux_init();
+
+  irq_global_ctrl(true);
+  irq_timer_ctrl(true);
+
+  mmio_region_t timer_reg =
+      mmio_region_from_addr(TOP_EARLGREY_RV_TIMER_BASE_ADDR);
+  CHECK(dif_rv_timer_init(
+            timer_reg,
+            (dif_rv_timer_config_t){.hart_count = 1, .comparator_count = 1},
+            &timer) == kDifRvTimerOk);
+  dif_rv_timer_tick_params_t tick_params;
+  CHECK(dif_rv_timer_approximate_tick_params(kClockFreqHz, kTickFreqHz,
+                                             &tick_params) ==
+        kDifRvTimerApproximateTickParamsOk);
+  CHECK(dif_rv_timer_set_tick_params(&timer, kHart, tick_params) ==
+        kDifRvTimerOk);
+  CHECK(dif_rv_timer_irq_enable(&timer, kHart, kComparator,
+                                kDifRvTimerEnabled) == kDifRvTimerOk);
+
+  dif_gpio_config_t gpio_config = {.base_addr =
+                                       mmio_region_from_addr(0x40010000)};
+  CHECK(dif_gpio_init(&gpio_config, &gpio) == kDifGpioOk);
```

Comment:
Sorry about this, it should ideally take config by value instead of pointer. I should fix this soon.